### ISSUE TITLE
Codegen does not generate generic List

### ISSUE THREAD

**Issue by komo81** [swagger-api/swagger-codegen#473]
Hi, 

In my api model I have:
`@JsonProperty("repriceItems") private List<RepriceItem> items;`
swagger json (I think correct):
`"repriceItems": { "items": { "type": "RepriceItem" }, "required": false, "type": "List" }`

But when I run codegen for Java or Android, I get:
`private List repriceItems = null;`
instead of
`private List<RepriceItem> repriceItems = null;`

I'm using codegen v 2.1.2-M1.

Is it a bug or do I something wrong?

**Comment by webron** [comment]
@komo81 - what do you use to produce your spec?

**Comment by komo81** [comment]
https://github.com/martypitt/swagger-springmvc v 0.9.5

**Comment by webron** [comment]
That's what I suspected. This is related to https://github.com/martypitt/swagger-springmvc/issues/593. Unfortunately, that's not a valid spec so the codegen doesn't process it properly. We should probably make it more robust though to deal with it.

**Comment by komo81** [comment]
Ok, so what should I do? Is it some workaround for it?

**Comment by webron** [comment]
You can change it manually and run the codegen against it (that's what the author of that issue has done). I may be able to point you to the LoC in the codegen so that you can change that locally, compile it, and use that instead.

**Comment by komo81** [comment]
ok, can you please point to the LoC?

**Comment by webron** [comment]
iirc, it's https://github.com/swagger-api/swagger-codegen/blob/master/modules/swagger-codegen/src/main/java/com/wordnik/swagger/codegen/DefaultCodegen.java#L193 - you can either change it or add another one for "List". Let me know if it works for you.

**Comment by komo81** [comment]
I added 
`typeMapping.put("List", "array");`
but now it generates 
`private Array repriceItems = null;`
so it's not working. I tried to debug the code and I think problem is in SwaggerParser class because property is parsed as RefProperty instead of ArrayProperty and the inner class type information is lost, so codegen cannot generate this property properly. 
So I think only possilble solutions are either fix swagger-springmvc to print List as 'array' in spec or fix SwaggerParser to read List property as ArrayProperty. Can you confirm that?

**Comment by fehguy** [comment]
Please post a more complete example of your swagger json or it's impossible to help.

**Comment by webron** [comment]
To be honest, I'm a bit swamped right now so diving into this issue is not an option for me. In any case, you should add your comment to that issue on swagger-springmvc otherwise they'll have no way of knowing other people suffer from the issue and it may not be prioritized.

**Comment by webron** [comment]
@fehguy - we've had this issue before. swagger-springmvc produces arrays using `"type": "List"` instead of `"type": "array"`.

**Comment by fehguy** [comment]
Yes, I do know that but I really need a complete example of what it's generating before spending the time on helping with a work around.  I do want to provide some special handling in swagger-compat-parser to support swagger-spring-mvc because this defect from that project causes a huge amount of support from us.

So @komo81 if you can anonymize your spec and share it, I will see if we can work around it in our side.

**Comment by komo81** [comment]
Here is my complete spec http://pastebin.com/LpSeNrKQ. Problem is only in this property of Reprice object:
` "repriceItems":
               {
                   "items":
                   {
                       "type": "RepriceItem"
                   },
                   "required": false,
                   "type": "List"
               },`

I tried to manually change `"type": "List"` to `"type": "array"` and then the codegen generated correctly `List<RepriceItem> items;`

**Comment by fehguy** [comment]
Thanks @komo81 I'll look in the next few hours.  You'll certainly have to build from develop_2.0 and run off that when I have the fix in.

**Comment by komo81** [comment]
Ok, I will wait :) Thank you very much.

**Comment by webron** [comment]
This is affected by https://github.com/swagger-api/swagger-parser/issues/28.

**Comment by carlin-q-scott** [comment]
The link showing how this could be fixed locally no longer works and I'm not understanding the workaround anyways. 

Here's my model:
```
{
    "id": "CreateCustomerRole",
    "properties": {
        "CustomerId": {
            "type": "string",
            "required": true
        },
        "RoleName": {
            "type": "string",
            "required": true
        },
        "Permissions": {
            "type": "Array",
            "items": {
                "type": "string"
            },
            "required": true
        },
        "Users": {
            "type": "Array",
            "items": {
                "type": "long"
            },
            "required": true
        }
    }
}
```

Here's the C# generated model from that:
```
/// <summary>
/// Gets or Sets CustomerId
/// </summary>
[DataMember(Name="CustomerId", EmitDefaultValue=false)]
public string CustomerId { get; set; }
/// <summary>
/// Gets or Sets RoleName
/// </summary>
[DataMember(Name="RoleName", EmitDefaultValue=false)]
public string RoleName { get; set; }
/// <summary>
/// Gets or Sets Permissions
/// </summary>
[DataMember(Name="Permissions", EmitDefaultValue=false)]
public List Permissions { get; set; }
/// <summary>
/// Gets or Sets Users
/// </summary>
[DataMember(Name="Users", EmitDefaultValue=false)]
public List Users { get; set; }
```

I have multiple warnings in the codegen output but it seems to all be related to a different issue involving empty models for Options requests.

**Comment by wing328** [comment]
It should be `"type": "array"` (lower case). (e.g. https://github.com/swagger-api/swagger-codegen/blob/master/modules/swagger-codegen/src/test/resources/2_0/petstore.json#L893)
